enterprise/dev/ci/internal/ci/pipeline.go

```diff
@@ -237,7 +237,12 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
 		// Slow image builds
 		imageBuildOps := operations.NewNamedSet("Image builds")
 		for _, dockerImage := range images.SourcegraphDockerImages {
-			imageBuildOps.Append(buildCandidateDockerImage(dockerImage, c.Version, c.candidateImageTag()))
+			// Only upload sourcemaps for the "frontend" image, on the Main branch build
+			uploadSourcemaps := false
+			if c.RunType.Is(runtype.MainBranch) && dockerImage == "frontend" {
+				uploadSourcemaps = true
+			}
```

Comment:
nit: you can remove the unnecessary if statement here!

```suggestion
			uploadSourcemaps := c.RunType.Is(runtype.MainBranch) && dockerImage == "frontend"
```

Reply:
Ahh that's so much more satisfying to look at!